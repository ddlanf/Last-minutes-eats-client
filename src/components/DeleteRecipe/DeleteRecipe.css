.delete-recipe-form{ 
    margin-left: auto; 
    margin-right: auto; 
    display: block; 
    position: fixed;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
    height:100%;
    padding-top: 20%;
    top: -900px;
    left: 0;
    z-index:1000;
    background-color: rgba(0, 45, 82, 0.438);
    text-align: center; 
    animation-name: slide-down;
    animation-fill-mode: forwards;
    animation-duration: 0.5s;
    animation-timing-function: ease-in;
}

@keyframes slide-down{
    0% { top: -900px}
    90%{ top: 0}
    92%{ top: 30px }
    94%{ top: 0px }
    96%{ top: 20px }
    98%{ top: 0}
    99%{ top: 10px }
    100%{ top: 0}
}

.delete-recipe-label{
    font-size: 7vw;
    margin-bottom: 2vw;
    display: block;
    width: 80%;
    margin-left: auto;
    margin-right: auto;
}

.delete-recipe-input{
    font-size: 6vw;
    width: 50%;
    max-width: 800px;
}

.delete-recipe-submit{
    cursor: pointer;
    display: block;
    margin-left: auto;
    margin-right: auto;
    background-color: transparent;
    border: 1.5px solid #209463;
    border-radius: 3rem;
    padding: .5rem 2rem;
    color:  #209463;
    font-size: 6vw;
    margin-top: 5%;
}

a.delete-recipe-back{
    display: block;
    margin-left: auto;
    margin-right: auto;
    text-align: center;
    background-color: transparent;
    border: 1.5px solid rgb(99, 99, 99);
    color: rgb(99, 99, 99);
    width: 25%;
    text-decoration: none;
    border-radius: 3rem;
    font-size: 6vw;
    margin-top: 5%;
}


.delete-recipe-error{
    color: red;
    font-size: 5vw;
    margin-top: 2%;
}

.delete-recipe-submitted{
    width: 90%;
    margin-left: auto;
    margin-right: auto;
    font-size: 4.2vw;
}

@media screen and (min-width: 1000px){
    .delete-recipe-label{
        font-size: 4vw;
        margin-top: 10vw;
        margin-bottom: 5%;
    }

    .delete-recipe-input{
        font-size: 4vw;
    }

    .delete-recipe-link{
        margin-top: 5%;
        font-size: 3vw;
    }

    .delete-recipe-submit{
        font-size: 3vw;
        margin-top: 3%;
    }

    .delete-recipe-form{
        padding-top: 8%;
    }

    .delete-recipe-error{
        font-size: 3vw;
    }

    .delete-recipe-submitted{
        font-size:  3vw;
        width: 100%;
    }

    a.delete-recipe-back{
        font-size: 3.5vw;
        width: 13%;
    }
}