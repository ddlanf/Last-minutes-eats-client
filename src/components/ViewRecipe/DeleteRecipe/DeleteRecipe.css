.delete-recipe-form{ 
    margin-left: auto; 
    margin-right: auto; 
    display: block; 
    position: fixed;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
    height:100%;
    padding-top: 20%;
    top: -900px;
    left: 0;
    z-index:1000;
    background-color: rgba(0, 45, 82, 0.616);
    text-align: center; 
    animation-name: slide-down;
    animation-fill-mode: forwards;
    animation-duration: 0.5s;
    animation-timing-function: ease-in;
}

.slide-up{
    animation-name: slide-up;
    animation-duration: 0.5s;
    animation-fill-mode: forwards;
    animation-timing-function: linear;
}

@keyframes slide-down{
    0% { top: -900px}
    90%{ top: 0}
    92%{ top: 30px }
    94%{ top: 0px }
    96%{ top: 20px }
    98%{ top: 0}
    99%{ top: 10px }
    100%{ top: 0}
}

@keyframes slide-up{
    0% { top: 0}
    100%{ top: -900px; display: none; }
}

.delete-recipe-label{
    font-size: 7vw;
    margin-bottom: 2vw;
    display: block;
    width: 80%;
    margin-left: auto;
    margin-right: auto;
    color: white;
}

.delete-recipe-input{
    font-size: 6vw;
    width: 50%;
    max-width: 800px;
}

.delete-recipe-submit{
    cursor: pointer;
    display: block;
    margin-left: auto;
    margin-right: auto;
    background-color: #209463;
    border: 1.5px solid #209463;
    color: white;
    border-radius: 3rem;
    padding: .5rem 2rem;
    font-size: 6vw;
    margin-top: 5%;
    width: 40%;
    max-width: 350px;
}

.delete-recipe-back{
    cursor: pointer;
    display: block;
    text-decoration: none;
    margin-left: auto;
    margin-right: auto;
    background-color: rgb(99, 99, 99);
    border: 1.5px solid rgb(99, 99, 99);
    color: white;
    border-radius: 3rem;
    padding: 1%;
    font-size: 6vw;
    margin-top: 5%;
    width: 32%;
    max-width: 300px;
}


.delete-recipe-error{
    color: red;
    font-size: 5vw;
    background-color: #002D52;
    width: 50%;
    border-radius: 30%;
    margin-right: auto;
    margin-left: auto;
    margin-top: 2%;
}

.delete-recipe-submitted{
    width: 90%;
    margin-left: auto;
    margin-right: auto;
    font-size: 4.2vw;
}

@media screen and (min-width: 550px){
    .delete-recipe-form{
       padding-top: 10%;
    }
}


@media screen and (min-width: 1000px){

    .delete-recipe-form{
        padding-top: 7%;
     }

    .delete-recipe-label{
        font-size: 5vw;
        margin-top: 6vw;
        margin-bottom: 5%;
    }

    .delete-recipe-input{
        font-size: 5vw;
    }

    .delete-recipe-submit{
        font-size: 70px;
    }

    .delete-recipe-back{
        font-size: 70px;
        padding: 0.5%;
    }

    .delete-recipe-error{
        font-size: 3vw;
    }

    .delete-recipe-submitted{
        font-size:  3vw;
        width: 100%;
    }

}