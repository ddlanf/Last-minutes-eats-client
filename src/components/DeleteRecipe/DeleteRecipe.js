import React, { Component } from 'react'
import { Link, withRouter} from 'react-router-dom' 
import './DeleteRecipe.css'
import RecipesApiService from '../../services/recipes-api-service'

class DeleteRecipe extends Component {

    constructor(props){
        super(props);
        this.state = { 
            token: '',
            submitted: false,
            error: ''
        }
    }

    handleInputChange = (event) =>{
        const { name } = event.target
        const { value } = event.target
        
        this.setState({
            [name] : value
        })
    }

    deleteRecipe = (ev) =>{
        ev.preventDefault()

        const { recipeId } = this.props.match.params;
        const { token } = this.state

        if(!token){
            this.setState({ error: 'Please enter token'})
        }
 
        else{
          RecipesApiService.deleteRecipe(recipeId, token)
            .then(res =>{
                this.setState({ submitted: true})
            })
            .catch((res)=>{ this.setState({ error: res.error })})
        }
    }

    render() {

        const { recipeId } = this.props.match.params;

        return (
            <form
                onSubmit={this.deleteRecipe}
                className="delete-recipe-form">
                <label className="delete-recipe-label">
                    Enter token to delete this recipe
                </label>
                <input 
                    type="password"
                    name="token"
                    onChange={this.handleInputChange}
                    className="delete-recipe-input"/>
                {!this.state.submitted ? (this.state.error  ? <p className="delete-recipe-error">{this.state.error}</p> : '') : ''}
                {this.state.submitted ? 
                <p className="delete-recipe-submitted">
                    Recipe deleted
                </p> : ''}
                <button
                    disabled={this.state.submitted}
                    type="submit"
                    className="delete-recipe-submit"
                    >
                    Submit
                </button>
                {this.state.submitted ?
                    <Link
                        to={`/view-all-recipes`}
                        className="delete-recipe-link"
                        >
                        Back
                    </Link>
                    :
                    <Link
                        to={`/view-recipe/${recipeId}`}
                        className="delete-recipe-link"
                        >
                        Back
                    </Link>}
            </form>
        )
    }
}

export default withRouter(DeleteRecipe)