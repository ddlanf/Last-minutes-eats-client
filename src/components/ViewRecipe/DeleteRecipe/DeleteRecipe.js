import React, { Component } from 'react'
import { withRouter} from 'react-router-dom' 
import './DeleteRecipe.css'
import RecipesApiService from '../../../services/recipes-api-service'

class DeleteRecipe extends Component {

    constructor(props){
        super(props);
        this.state = { 
            token: '',
            submitted: false,
            error: '',
            back: false
        }
    }

    handleInputChange = (event) =>{
        const { name } = event.target
        const { value } = event.target
        
        this.setState({
            [name] : value
        })
    }

    deleteRecipe = (ev) =>{
        ev.preventDefault()

        const { recipeId } = this.props.match.params;
        const { token } = this.state

        if(!token){
            this.setState({ error: 'Please enter token'})
        }
 
        else{
          RecipesApiService.deleteRecipe(recipeId, token)
            .then(res =>{
                this.setState({ submitted: true})
            })
            .catch((res)=>{ this.setState({ error: res.error })})
        }
    }

    goBack = () =>{
        if(this.state.submitted){
            this.props.history.push('/view-all-recipes')
        }
        this.setState({ back: true })
    } 

    setDisplayProperty = () =>{
        if(this.state.back){ 
            this.props.toggleDeleteRecipe(false)
            this.setState({ back: false }) 
        }
    }

    render() {

        return (
            <form
                style={{ display: this.props.showDeleteRecipe ? 'block' : 'none'}}  
                onAnimationEnd={this.setDisplayProperty}
                onSubmit={this.deleteRecipe}
                className={`delete-recipe-form ${this.state.back ? 'slide-up' : ''}`}>
                <label className="delete-recipe-label">
                    Enter token to delete this recipe
                </label>
                <input 
                    type="password"
                    name="token"
                    onChange={this.handleInputChange}
                    className="delete-recipe-input"/>
                {!this.state.submitted ? (this.state.error  ? <p className="delete-recipe-error">{this.state.error}</p> : '') : ''}
                {this.state.submitted ? 
                <p className="delete-recipe-submitted">
                    Recipe deleted
                </p> : ''}
                <button
                    disabled={this.state.submitted}
                    type="submit"
                    className="delete-recipe-submit"
                    >
                    Submit
                </button>            
                <button
                    onClick={this.goBack}
                    type="button"
                    className="delete-recipe-back">                        
                    Back 
                </button>
            </form>
        )
    }
}

export default withRouter(DeleteRecipe)