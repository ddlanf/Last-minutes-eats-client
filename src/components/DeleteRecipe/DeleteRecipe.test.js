import React from 'react';
import ReactDOM from 'react-dom';
import DeleteRecipe from './DeleteRecipe';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'

describe('Make component', () => {
    

    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                        <DeleteRecipe/>
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(
                <BrowserRouter>
                    <DeleteRecipe/>
                </BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});