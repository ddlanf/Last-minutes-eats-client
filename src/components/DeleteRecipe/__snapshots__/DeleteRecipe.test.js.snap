// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make component renders the UI as expected 1`] = `
<form
  className="delete-recipe-form"
  onSubmit={[Function]}
>
  <label
    className="delete-recipe-label"
  >
    Enter token to delete this recipe
  </label>
  <input
    className="delete-recipe-input"
    name="token"
    onChange={[Function]}
    type="password"
  />
  
  
  <button
    className="delete-recipe-submit"
    disabled={false}
    type="submit"
  >
    Submit
  </button>
  <a
    className="delete-recipe-back"
    href="/view-recipe/undefined"
    onClick={[Function]}
  >
    Back
  </a>
</form>
`;
